src/ltablib.js

```diff
@@ -166,7 +169,7 @@ const unpack = function(L) {
     let e = lauxlib.luaL_opt(L, lauxlib.luaL_checkinteger, 3, lauxlib.luaL_len(L, 1));
     if (i > e) return 0;  /* empty range */
     let n = e - i;  /* number of elements minus 1 (avoid overflows) */
-    if (n >= llimit.MAX_INT || !lua.lua_checkstack(L, ++n))
+    if (n >= INT_MAX || !lua.lua_checkstack(L, ++n))
```

Comment:
I think this can be replaced with the actual js largest integer (`Number.MAX_SAFE_INTEGER`)